core/src/databases/table_schema.rs

```diff
@@ -262,6 +265,85 @@ impl TableSchema {
                 .collect::<Result<Vec<_>>>(),
         }
     }
+
+    pub fn merge(&self, other: &Self) -> Result<Self, anyhow::Error> {
```

Comment:
I briefly tried to make this happen in-place on a mutable ref, but it turns out it was a bit painful and made the API a bit annoying to use (everything needs to be mutable, which forced me to create a bunch of `*_mut()` methods which I didn't like.
I don't think creating a new `TableSchema` is much of an issue here.

Reply:
You should never need `_mut()` methods but like this is fine anyway